reports/GitalyFail/GitalyFail.tla

```diff
@@ -0,0 +1,137 @@
+
+\* Don't call - endless loop
+TMSend(r) == /\ \E r \in RM
+             /\ rmState[r] = "working"
+             /\ RMRcvPrep(r)
+             /\ UNCHANGED << rmState, tmState, msgs >> 
+
+(*
+    Expected stages 
+    
+    TMInit + RMInit (all)
+    TMsendPrepare
+    RMRcvPrep -
+        RMIgnorePrep
+        RMDoPrepare
+            RMConfirmPrepare - TMRcvConfirmation
+    TMCalculateQuorum - add rms to voted
+    TMSendCommit - to voted rms
+    TMSendAbort - to voted rms
+    RMRvcAbort
+    RMRcvCommit -
+        RMIgnoreCommit
+        RMDoCommit
+            RMConfirmCommit - add rms to confirmed commits
+    TMIncreaseGen for confirmed
+    
+*)
+
+DoNothing == /\ UNCHANGED << tmState, rmState >>
+
+RMRcvPrep(r) == /\ \E r \in RM 
+                /\ rmState[r] == "worknig"
+                /\ (\/ DoNothing
+                    \/ RMDoPrepare(r) )
+                                      
+RMDoPrepare(r) ==  /\ rmState' = [rmState EXCEPT ![r] = "prepared"]
+\*                /\ msgs' = msgs \union {[type |-> "Prepared", rm |-> r]}
+                   /\ RMConfirmPrepare(r)
+                   /\ UNCHANGED << tmState, tmPrepared >>
+
+RMConfirmPrepare(r) == \/ DoNothing
+                       \/ TMConfirmPrepare(r) 
+
+TMConfirmPrepare(r) == /\ rmPrepared' = rmPrepared \union r
+                       /\ msgs' = msgs \union {[type |-> "Prepared", rm |-> r]}
+        
+IsQuorumPrepared == /\ [type |-> "Abort"] \notin msgs
+                    /\ Cardinality(msgs) * 2 > Cardinality(Nodes)
+                    \* AND MasterNode voted for Commit                     
+
+TMSendCommit == /\ IsQuorumPrepared
+                /\ tmState = "init"
+                /\ tmState' = "done"
+                \* /\ tmPrepared = RM
+                /\ msgs' =  msgs \union {[type |-> "Commit"]}
+                /\ tmGeneration' = tmGeneration + 1
+                /\ UNCHANGED << rmState, tmPrepared >>  
+
+TMSendAbort ==  /\ tmState = "init"
+                /\ tmState' = "done"
+                /\ msgs' =  msgs \union {[type |-> "Abort"]}
+                /\ UNCHANGED << rmState, tmPrepared >>  
+
+\* Here it is possible to devide to DoNothing and DoAbort, but it does not matter for the inconsistency issue. - ???
+RMRcvAbort(r) == /\ rmState[r] \= "committed"
+                 /\ [type |-> "Abort"] \in msgs
+                 /\ rmState' = [rmState EXCEPT ![r] = "aborted"]
+                 /\ UNCHANGED << tmState, tmPrepared, msgs >>
+                
+RMRcvCommit(r) == /\ [type |-> "Commit"] \in msgs  
+                  /\ (\/ DoNothing                
+                      \/ RMDoCommit(r))
+                  /\ UNCHANGED << tmState, tmPrepared, msgs >> 
+                
+RMDoCommit(r) == /\ rmState' = [rmState EXCEPT ![r] = "committed"]
+                 /\ (\/ DoNothing
+                     \/ RMConfirmCommit(r) )
+
+\* In fact  - TM got message rm[i] committed
+RMConfirmCommit(r) == /\ msgs' = msgs \union {[type |-> "Commit", rm |-> r]}
```

Comment:
Maybe change `type` of message to `commited` to avoid collision between TM and RM messages?

Reply:
Good point, right.